### ISSUE TITLE
Cannot train: 'tuple' object has no attribute 'eval'

### ISSUE THREAD

**Issue by wichtounet** [hunkim/word-rnn-tensorflow#9]
Hi, 

I tried this implementation of word-rnn and I have an issue with the train script: 

> Traceback (most recent call last):
>   File "train.py", line 111, in <module>
>     main()
>   File "train.py", line 48, in main
>     train(args)
>   File "train.py", line 93, in train
>     state = model.initial_state.eval()
> AttributeError: 'tuple' object has no attribute 'eval'

I'm using the last version of tensorflow and Python 3.4. 

I've already had this issue with one other tensorflow project so I guesss there must be something wrong with my installation. 

Thanks

**Comment by moonlightlane** [comment]
I got exactly the same problem with Python 2.7. I think the problem comes from `initial_tuple` which is organized as a tuple. Previously it is not a tuple.

**Comment by hunkim** [comment]
Thanks! Do you know which version of TF did you use?

**Comment by moonlightlane** [comment]
I think it is 0.10. There are some posts on github about this issue but none of them work for me... `model.initial_state` looks like this: `(LSTMStateTuple(c=<tf.Tensor 'zeros:0' shape=(50, 128) dtype=float32>, h=<tf.Tensor 'zeros_1:0' shape=(50, 128) dtype=float32>), LSTMStateTuple(c=<tf.Tensor 'zeros_2:0' shape=(50, 128) dtype=float32>, h=<tf.Tensor 'zeros_3:0' shape=(50, 128) dtype=float32>))
`, and it seems that the state is a tuple of tuples. I don't yet have a good idea how to deal with it.

**Comment by hunkim** [comment]
Let me try with 0.10. Thanks!

**Comment by astr010** [comment]
I am using Ubuntu/Linux 64-bit CPU Only, Python 2.7 Tensorflow-0.11 and the work around in the issue below fixed this issue for me
https://github.com/sherjilozair/char-rnn-tensorflow/issues/47

TL;DR
in train.py try replacing the line
`state = model.initial_state.eval()`
with 
`model.initial_state = tf.convert_to_tensor(model.initial_state) state = model.initial_state.eval()`

**Comment by hunkim** [comment]
Can you write a pull request for us?

Sung

**Comment by wichtounet** [comment]
I made a PR but using sess.run(x) instead of convert_to_tensor and then eval. I tested it on two machines one with standard installation and one with CUDA installation (the one with the original issue).

**Comment by hunkim** [comment]
Please reopen if this issue reoccurs.
